app/api/files/routes.ts

```
@@ -221,15 +271,47 @@ export default (app: Application) => {
         filename: req.params.filename,
       });
 
+      const currentUser = permissionsContext.getUserInContext();
+
       if (
         !file?.filename ||
         !file?.type ||
         !(await storage.fileExists(file.filename, file.type)) ||
-        !(await checkEntityPermission(file, permissionsContext.getUserInContext()))
+        !(await checkEntityPermission(file, currentUser))
       ) {
         throw createError('file not found', 404);
       }
 
+      // Set cache control and Last-Modified headers (only if feature flag is enabled)
+      // Wrapped in try-catch to ensure this never blocks file serving
+      try {
+        if (tenants.current().featureFlags?.fileCacheHeaders) {
+          // Optimization: Authenticated users always get private cache, skip complex logic
+          if (currentUser) {
+            res.setHeader('Cache-Control', 'private, max-age=3600');
+          } else {
+            // Unauthenticated access - determine if publicly cacheable
+            const appSettings = await settings.get();
+            const isPrivateInstance = appSettings.private || false;
+
+            // Check if file is publicly accessible
+            const isPublic = await isFilePubliclyAccessible(file, isPrivateInstance);
+
+            const cacheControl = getCacheControlHeader(isPublic, isPrivateInstance);
+            res.setHeader('Cache-Control', cacheControl);
+          }
+
+          // Set Last-Modified header
+          if (file.creationDate) {
+            const lastModified = timestampToHTTPDate(file.creationDate);
+            res.setHeader('Last-Modified', lastModified);
+          }
+        }
+      } catch (error) {
+        // If any cache header logic fails, silently skip it and continue serving file
+        // This ensures backwards compatibility and prevents blocking file access
+      }
```

Again a catch block failing silently

---

Yes, lets not leave it like this please

> // This ensures backwards compatibility and prevents blocking file access

What this makes sure of is to hide potential important errors, let the error bubble up, in this case it is probably enough to remove the try/catch block